### PR TITLE
feat: Add GitProvenance support for Azure DevOps repositories

### PR DESCRIPTION
**Pull request by bryceatmoderne**
## What's changed?
Updated `GitProvenance` to support parsing Azure DevOps repositories.

## What's your motivation?
These changes are required to support Azure DevOps as an SCM.

## Anything in particular you'd like reviewers to focus on?
Let me know of other `rewrite` classes that may require similar updates.

## Anyone you would like to review specifically?
@timtebeek 

## Have you considered any alternatives or workarounds?
No.

## Any additional context
No.

### Checklist
- [ ] I've added unit tests to cover both positive and negative cases
- [ ] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [ ] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: rewrite-core/src/test/java/org/openrewrite/marker/GitProvenanceTest.java [hunk 1]
```diff
@@ -83,9 +91,9 @@ void getOrganizationNameWithBaseUrl(String gitOrigin, String baseUrl, String org
 
     @ParameterizedTest
     @MethodSource("remotes")
-    void getRepositoryName(String remote) {
+    void getRepositoryName(String remote, String org, String repo) {
```
Comment (timtebeek) [inline rewrite-core/src/test/java/org/openrewrite/marker/GitProvenanceTest.java:94]: Can we collapse this and `getOrganizationName` above to get rid of the unused argument warnings?

**Review by timtebeek — APPROVED** [review]
This has always been a bit of an oddity; especially the long deprecated but never fully removed `getOrganizationName()`. Thanks for extending it thoughtfully.

File: rewrite-core/src/test/java/org/openrewrite/marker/GitProvenanceTest.java (hunk 1 above)
Comment (bryceatmoderne) [reply rewrite-core/src/test/java/org/openrewrite/marker/GitProvenanceTest.java]: Done. Thanks for the suggestion!

**Comment by bryceatmoderne** [comment]
This requirement was addressed by https://github.com/openrewrite/rewrite/pull/4381

**Pull request closed without merging**